src/robot/client.hpp

```diff
@@ -86,3 +105,26 @@ class RobotClient {
     ResourceManager resource_manager;
     void refresh_every();
 };
+
+template <typename T>
+std::shared_ptr<T> typed_resource_from_robot(const std::shared_ptr<RobotClient> client,
```

Comment:
This feels like it should be a member of `RobotClient`. I think it could even be overloaded as `resource_by_name` because it has different arguments:

```
class RobotClient {
pubilc:
    std::shared_ptr<ResourceBase> resource_by_name(ResourceName name, ResourceType type = ...);

    template<typename T>
    std::shared_ptr<T> resource_by_name(const std::string& name);
```